controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java

```diff
@@ -69,16 +71,66 @@ public boolean authenticate(String resource, String credentials, AuthHandler.Per
             String token = parts[1];
             AuthHandler handler = getHandler(method);
             assert handler != null;
-            if (!handler.authenticate(token)) {
+            Principal principal;
+            if ((principal = handler.authenticate(token)) == null) {
                 throw new AuthenticationException("Authentication failure");
             }
-            retVal = handler.authorize(resource, token).ordinal() >= level.ordinal();
+            retVal = handler.authorize(resource, principal).ordinal() >= level.ordinal();
         } catch (RuntimeException e) {
             throw new AuthenticationException(e);
+        } catch (AuthException e) {
+            throw new AuthenticationException("Authentication failure");
         }
         return retVal;
     }
 
+    /**
+     *
+     * API to authenticate a given credential.
+     * @param credentials  Credentials used for authentication.
+     * @return         Returns the Principal if the entity represented by credentials is authenticated.
+     * @throws AuthException if an authentication failure occurred.
+     */
+    public Principal authenticate(String credentials) throws AuthException {
+        Preconditions.checkNotNull(credentials, "credentials");
+        boolean retVal = false;
+        String[] parts = credentials.split("\\s+", 2);
+        if (parts.length != 2) {
+            throw new AuthenticationException("Malformed request");
+        }
+        String method = parts[0];
+        String token = parts[1];
+        AuthHandler handler = getHandler(method);
+        assert handler != null;
+        return handler.authenticate(token);
+    }
+
+    /**
+     *
+     * API to authorize a given principal and credential.
+     *
+     * @param resource The resource identifier for which the access needs to be controlled.
```

Comment:
Please align the `@param` texts; the IDE should do it for you

Reply:
Fixed. IDE did not do it for me :). Did by hand.